kornia/augmentation/_2d/mix/transplantation.py

```diff
@@ -0,0 +1,261 @@
+                 [2, 1, 0, 0, 0]]])
+        >>> image_out, mask_out = aug(image, mask)
+        >>> image_out.shape
+        torch.Size([2, 3, 5, 5])
+        >>> mask_out.shape
+        torch.Size([2, 5, 5])
+        >>> mask_out
+        tensor([[[2, 0, 1, 2, 0],
+                 [2, 2, 0, 2, 0],
+                 [2, 2, 1, 1, 2],
+                 [2, 2, 2, 0, 2],
+                 [2, 2, 2, 0, 2]],
+        <BLANKLINE>
+                [[0, 0, 0, 2, 0],
+                 [2, 1, 0, 0, 0],
+                 [2, 0, 1, 0, 0],
+                 [0, 0, 0, 0, 2],
+                 [2, 1, 0, 0, 0]]])
+
+    You can apply the same augmentation again, but the new mask should contain the same labels for the augmentation
+    to have an effect:
+
+        >>> aug._params["selected_labels"]  # Image 0 received label 2 from image 1 and image 1 label 0 from image 0
+        [tensor(2), tensor(0)]
+        >>> mask2 = torch.randint(0, 3, (2, 5, 5))
+        >>> image_out2, mask_out2 = aug(image, mask2, params=aug._params)
+        >>> mask_out2
+        tensor([[[2, 2, 2, 0, 1],
+                 [1, 2, 0, 1, 1],
+                 [0, 0, 1, 2, 2],
+                 [0, 0, 1, 2, 1],
+                 [1, 1, 1, 2, 0]],
+        <BLANKLINE>
+                [[0, 2, 2, 0, 0],
+                 [0, 0, 0, 0, 1],
+                 [0, 0, 0, 2, 0],
+                 [0, 0, 1, 0, 1],
+                 [1, 1, 1, 2, 0]]])
+    """
+
+    def __init__(
+        self,
+        excluded_labels: Optional[Sequence[int]] = None,
+        p: float = 0.5,
+        p_batch: float = 1.0,
+        data_keys: Optional[List[Union[str, int, DataKey]]] = None,
+    ) -> None:
+        super().__init__(p=p, p_batch=p_batch)
+
+        if excluded_labels is None:
+            excluded_labels = []
+        self.excluded_labels = tensor(excluded_labels)
+        if self.excluded_labels.ndim != 1:
+            raise ValueError(
+                f"excluded_labels must be a 1-dimensional sequence, but got {self.excluded_labels.ndim} dimensions."
+            )
+
+        if data_keys is None:
+            data_keys = [DataKey.INPUT, DataKey.MASK]
+        self.data_keys = [DataKey.get(inp) for inp in data_keys]
+        self._channel_dim = 1
+
+    def apply_non_transform_mask(self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any]) -> Tensor:
+        return input
+
+    def transform_input(self, acceptor: Tensor, donor: Tensor, selection: Tensor) -> Tensor:
+        # Expand selection to the channel dimension
+        selection = selection.unsqueeze(dim=self._channel_dim).expand_as(donor)
+        acceptor[selection] = donor[selection]
+        return acceptor
+
+    def transform_mask(self, acceptor: Tensor, donor: Tensor, selection: Tensor) -> Tensor:
+        acceptor[selection] = donor[selection]
+        return acceptor
+
+    def forward(  # type: ignore[override]
+        self,
+        image: Tensor,
+        mask: Tensor,
+        *additional_inputs: Tensor,
```

Comment:
Emm... This one is kinda a special case of input. We definitely cannot use it for `AugmentationSequential` if so.

Ideally, only image and mask are required for this forward here. Then the sequential processing for `additional_inputs` shall be supported by `AugmentationSequential`.

I am thinking that, removing the additional_inputs here. Then `AugmentationSequential.__init__` detects if we have a `transplantation`, then throws out a warning that `The first IMAGE and MASK key will be used for transplantation augmentation`. Then we make a special case to use both image and mask for the first round. Then everything else shall be as same as other augmentations, right?

Reply:
This is true. I removed the additional arguments and used the same API as the base class with only one `input` argument. I am now using the `data_keys` to identify the first mask tensor and that tensor determines the transplantation which is used for all tensors. This is more flexible now :-) 

But I do see the problem with `AugmentationSequential` now. This does indeed not work since per default only one input at a time is passed on to the transformations. I will try to find a solution for that.